### PR TITLE
Enable prebuilt detection

### PR DESCRIPTION
**Pull request by mmitche**
- Add some missing depenencies with source-build tags
- Update baseline
- Remove some unused NuGet properties
- Remove old source build workaround
- Remove explicit reference to Microbuild.Core
- Pass DotNetBuildFromSource to the repo tasks inner build environment

# {PR title}

- [ ] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/main/CODE-OF-CONDUCT.md).
- [ ] You've included unit or integration tests for your change, where applicable.
- [ ] You've included inline docs for your change, where applicable.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

Summary of the changes (Less than 80 chars)

## Description

{Detail}

Fixes #{bug number} (in this specific format)

### CODE REVIEW

File: eng/SourceBuildPrebuiltBaseline.xml [hunk 1]
```diff
@@ -1,5 +1,27 @@
 <UsageData>
   <IgnorePatterns>
-    <UsagePattern IdentityGlob="*/*" />
+    <UsagePattern IdentityGlob="Microsoft.SourceBuild.Intermediate.*/*" />
+
+    <!-- These are used in-build for repo-tasks and should come from previously source built -->
+    <UsagePattern IdentityGlob="NuGet.Common/*6.2.2*" />
```
Comment (wtgodbe) [inline eng/SourceBuildPrebuiltBaseline.xml:6]: Will we have to maintain these hardcoded versions?

Comment (wtgodbe) [reply eng/SourceBuildPrebuiltBaseline.xml:6]: Why not use the versions from `Versions.props`?

File: eng/tools/RepoTasks/RepoTasks.csproj [hunk 2]
```diff
@@ -23,6 +23,8 @@
     <PackageReference Include="Microsoft.Build.Framework" Version="$(MicrosoftBuildFrameworkVersion)" />
     <PackageReference Include="Microsoft.Build.Tasks.Core" Version="$(MicrosoftBuildTasksCoreVersion)" />
     <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(MicrosoftBuildUtilitiesCoreVersion)" />
+    <!-- Manually updated version from 6.0.0 to address CVE-2021-43877 -->
+    <PackageReference Include="System.Security.Cryptography.Xml" Version="6.0.1" />
```
Comment (wtgodbe) [inline eng/tools/RepoTasks/RepoTasks.csproj:27]: We should property-ize this and put it in `Versions.props`

File: eng/SourceBuildPrebuiltBaseline.xml (hunk 1 above)
Comment (mmitche) [reply eng/SourceBuildPrebuiltBaseline.xml:6]: This isn't loaded by msbuild. They will need to be maintained, but I assume they will not update too often.

File: eng/tools/RepoTasks/RepoTasks.csproj (hunk 2 above)
Comment (mmitche) [reply eng/tools/RepoTasks/RepoTasks.csproj]: Done. It differs from the 8.0 version so I chose a different property name

**Comment by mmitche** [comment]
@wtgodbe Looking good now.

File: eng/SourceBuildPrebuiltBaseline.xml (hunk 1 above)
Comment (wtgodbe) [reply eng/SourceBuildPrebuiltBaseline.xml:6]: Will we see an error if we update the related package versions but don't update these? Want to make sure we don't let it fall behind

**Review by wtgodbe — APPROVED** [review]

**Pull request merged**